src/com/facebook/buck/jvm/kotlin/KotlincToJarStepFactory.java

```diff
@@ -83,51 +128,244 @@ public void createCompileStep(
-                          .orElse(ImmutableList.of()))
-                  .addAll(declaredClasspathEntries)
-                  .build(),
+              allClasspaths,
               kotlinc,
-              extraArguments,
-              projectFilesystem));
+              ImmutableList.<String>builder()
+                  .addAll(extraArguments)
+                  .add(NO_STDLIB)
+                  .add(NO_REFLECT)
+                  .add(COMPILER_BUILTINS)
+                  .add(LOAD_BUILTINS_FROM)
+                  .add(VERBOSE)
+                  .add()
+                  .build(),
+              projectFilesystem,
+              Optional.of(parameters.getWorkingDirectory())));
     }
 
+    ImmutableSortedSet<Path> sources =
+        ImmutableSortedSet.<Path>naturalOrder()
+            .add(genOutput)
+            .addAll(sourceFilePaths)
+            .build();
+
     ImmutableSortedSet<Path> javaSourceFiles =
         ImmutableSortedSet.copyOf(
-            sourceFilePaths
+            sources
                 .stream()
-                .filter(JAVA_PATH_MATCHER::matches)
+                .filter(input -> !PathMatchers.KOTLIN_PATH_MATCHER.matches(input))
                 .collect(Collectors.toSet()));
 
-    // Don't invoke javac if we don't have any java files.
-    if (!javaSourceFiles.isEmpty()) {
-      CompilerParameters javacParameters =
-          CompilerParameters.builder()
-              .from(parameters)
-              .setClasspathEntries(
-                  ImmutableSortedSet.<Path>naturalOrder()
-                      .add(projectFilesystem.resolve(outputDirectory))
-                      .addAll(
-                          Optional.ofNullable(extraClassPath.getExtraClasspath())
-                              .orElse(ImmutableList.of()))
-                      .addAll(declaredClasspathEntries)
-                      .build())
-              .setSourceFilePaths(javaSourceFiles)
-              .build();
-      new JavacToJarStepFactory(
-              resolver, ruleFinder, projectFilesystem, javac, javacOptions, extraClassPath)
-          .createCompileStep(buildContext, invokingRule, javacParameters, steps, buildableContext);
-    }
+    CompilerParameters javacParameters =
+        CompilerParameters.builder()
+            .from(parameters)
+            .setClasspathEntries(
+                ImmutableSortedSet.<Path>naturalOrder()
+                    .add(projectFilesystem.resolve(outputDirectory))
+                    .addAll(
+                        Optional.ofNullable(extraClassPath.getExtraClasspath())
+                            .orElse(ImmutableList.of()))
+                    .addAll(declaredClasspathEntries)
+                    .build())
+            .setSourceFilePaths(javaSourceFiles)
+            .build();
+
+    new JavacToJarStepFactory(
+        resolver,
+        ruleFinder,
+        projectFilesystem,
+        javac,
+        javacOptions.withAnnotationProcessingParams(AnnotationProcessingParams.EMPTY),
```

Comment:
Internal CI is failing because annotation processors are not run on the Java parts of a mixed Java & Kotlin module, presumably because of this line.